Add support for Clipsal cent-a-meter CMR113 power meter

mikey opened the pull request:
Device only measures current but is only sold in Australia so driver
gives power output assuming voltage is 240V.

merbanan commented on src/devices/cmr113.c:
```diff
@@ -0,0 +1,110 @@
+    data = data_make(
+	    "model",   "",         DATA_STRING, "CMR112",
+	    "power_W", "Power",    DATA_FORMAT, "%.0f W", DATA_DOUBLE,  power,
+	    "current", "Current",  DATA_FORMAT, "%.1f A", DATA_DOUBLE,  current,
```
SI unit missing.

mikey replied:
SI missing on the current? should be "current_A" ?

merbanan replied:
I think that is the correct unit.

merbanan commented on src/devices/cmr113.c:
```diff
@@ -0,0 +1,110 @@
+    data = data_make(
+	    "model",   "",         DATA_STRING, "CMR112",
+	    "power_W", "Power",    DATA_FORMAT, "%.0f W", DATA_DOUBLE,  power,
+	    "current", "Current",  DATA_FORMAT, "%.1f A", DATA_DOUBLE,  current,
```
SI unit missing.

merbanan commented on examples/rtl_433_mqtt_relay.py:
```diff
@@ -18,7 +18,7 @@
 
-MQTT_HOST = "127.0.0.1"
+MQTT_HOST = "192.168.86.37"
 MQTT_PORT = 1883
```
Stray hunk

mikey replied:
oops, yes. Sorry

merbanan commented:
Can you add a bit stream description also? And then we need some sample signals also.

mikey commented:
> Can you add a bit stream description also? And then we need some sample signals also.

Yes will do. I have some good samples. I just need to take some photos of the device.

merbanan commented:
Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

mikey commented:
> Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

Fair enough. It would be nice to be able to add data about your configuration so we could do this. 

I'm also using the TS-FT002. Being able to add information about your water tank would enable us to broadcast the tank %. 

But maybe that's out of scope for this (already awesome) project :-)

mikey commented:
> Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

Would you also like me to keep the three current sensors separate, rather than aggregating them?

zuckschwerdt commented:
Convert tabs to 4 spaces, model should likely be `Clipsal-CMR112`, run `maintainer_update.py` to get e.g. example conf right.

merbanan commented:
> Would you also like me to keep the three current sensors separate, rather than aggregating them into one?

Not sure I follow you. Each transmitters data should be translated to a message. As direct translation as possible.

mikey commented:
> > Would you also like me to keep the three current sensors separate, rather than aggregating them into one?
> 
> Not sure I follow you. Each transmitters data should be translated to a message. As direct translation as possible.

OK. There is only 1 transmitter but it contains data for the 3 attached current sensors in each transmission. 

I'll output 3 separate current values (I'm just output the sum of the 3 currently).

zuckschwerdt commented:
> I'll output 3 separate current values (I'm just output the sum of the 3 currently).

Preferred keys then are: `current_1_A`, ...

mikey commented:
Pushed some updates:
1. 4 space tabs
2. Ran maintainer_update.py (BTW SoapySDR needs some updates according to it)
3. Remove stray hunk
4. Updated bitstream description 
5. Added SI unit to output
6. Removed power output and split current in 3 (current called current_1_A, current_3_A, current_3_A)
7. Updated model number

mikey commented:
@merbanan I've added some samples in https://github.com/merbanan/rtl_433_tests/pull/401

zuckschwerdt commented:
> samples in [merbanan/rtl_433_tests#401](https://github.com/merbanan/rtl_433_tests/pull/401)

Can you add back the freq/rate info to the filenames, it's the only metadata we have.

zuckschwerdt commented:
The `if` branch L55 needs more indent. (Clang warns about those things). `){` needs a space.
Indent on continued line, e.g. 97 107 117 is 8 spaces. (Try clang-format if you have Clang, it *mostly* gets things right.)

mikey commented:
> The `if` branch L55 needs more indent. (Clang warns about those things). `){` needs a space.
I understand these two and have fixed

> Indent on continued line, e.g. 97 107 117 is 8 spaces. (Try clang-format if you have Clang, it _mostly_ gets things right.)

I think I've fixed this.

zuckschwerdt commented:
Looks good, one thing I just notice: the "cmr112" should all be "cmr113", right?

mikey commented:
yes, sorry. I've updated the PR title now. I can't update the branch name without closing this PR. 

The code is correct and only refers to cmr113.

mikey commented:
> Branch name isn't an issue. But check the "Files changed" tab for the contents of the PR, there's 112 in multiple places.

Sorry, you're right. My bad, I 'git grepped' badly. Fixed now.

mikey commented:
I've updated the tests as well as that had the same mistake.

zuckschwerdt commented:
No worries. There is still one instance in example conf.
Also "model" needs to be "Clipsal-CMR113" (note the dash),
and output_fields needs to have: "model", "current_1_A", "current_2_A", "current_3_A", NULL.

mikey commented:
> No worries. There is still one instance in example conf.
:facepalm: Fixed

> Also "model" needs to be "Clipsal-CMR113" (note the dash),

OK

> and output_fields needs to have: "model", "current_1_A", "current_2_A", "current_3_A", NULL.

OK

I've repushed and also fixed up the tests to add the dash. Sorry for the noise.
